### PR TITLE
Adding first BLoC pattern implementation in the Roll the Dice App

### PR DESCRIPTION
**Pull request by HoracioDuca**
### The very first BLoC pattern implementation on the "Roll the Dice" App!

**This is an app in which you can tap on any dice and make all the dices roll!**

It has 6 dices but you can easily change the amount in the "Dimensions.dart" file.

Also you can see at the bottom of the screen a navigation bar that contains customizable texts and the score you get once you hit a dice!

Greetings :)

Animation

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+    super.initState();
+  }
+
+  final DiceBloc diceBloc = DiceBloc();
+
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Roll the Dice App',
+      title: Dimension.mainAppTitle,
```
Comment (marianosalvetti) [inline lib/main.dart]: it's fine, but maybe you need to rename the "Dimensions files" to "Constants, maybe.

File: lib/src/styles/dimensions.dart [hunk 2]
```diff
@@ -1,13 +1,22 @@
 abstract class Dimension {
```
Comment (marianosalvetti) [inline lib/src/styles/dimensions.dart]: maybe rename to Constants.

**Review by marianosalvetti — APPROVED** [review]

**Review by eis6aer — APPROVED** [review]
LGTM !

File: lib/main.dart (hunk 1 above)
Comment (HoracioDuca) [reply lib/main.dart]: done

File: lib/src/styles/dimensions.dart (hunk 2 above)
Comment (HoracioDuca) [reply lib/src/styles/dimensions.dart]: done

File: lib/src/bloc/dice_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    _streamDicesScoreController.close();
+  }
+
+  @override
+  void initialize() {}
+
+  @override
+  void roll() {
+    this._diceValues.clear();
+    int score = 0;
```
Comment (FranciscoGaraventa) [inline lib/src/bloc/dice_bloc.dart]: score is a private var of roll function, you need to add '_' at the beginning of the variable

File: lib/src/bloc/dice_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    for (int i = 0; i < Constants.diceCount; i++) {
+      int randomValue = Random().nextInt(Constants.diceFaces) + 1;
```
Comment (FranciscoGaraventa) [inline lib/src/bloc/dice_bloc.dart]: randomValue is the same like score, add '_' at beginning, and try to declare out of the loop. If you declare inside for you're creating a new variable for every loop

File: lib/src/bloc/dice_bloc.dart (hunk 3 above)
Comment (HoracioDuca) [reply lib/src/bloc/dice_bloc.dart]: done

File: lib/src/bloc/dice_bloc.dart (hunk 4 above)
Comment (HoracioDuca) [reply lib/src/bloc/dice_bloc.dart]: done. Thank you

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Pull request merged**